DEVELOPMENT.md

```diff
@@ -1,72 +1,167 @@
-## Install BentoML from source
+# Developer Guide
 
-Make sure to have [Git](https://git-scm.com/) and [Python3.7+](https://www.python.org/downloads/) installed.
+## Start Developing
 
-Optionally, make sure to have [GNU Make](https://www.gnu.org/software/make/) available on your system if you aren't using UNIX-based system for better developer experience.
-If you don't want to use `make` then refers to [Makefile](./Makefile) for specific commands on given make target.
+<details><summary><h3>with the Command Line</h3></summary>
 
-```bash
-python --version
+1. Make sure to have [Git](https://git-scm.com/), [pip](https://pip.pypa.io/en/stable/installation/), and [Python3.7+](https://www.python.org/downloads/) installed.
 
-pip --version
-```
+   Optionally, make sure to have [GNU Make](https://www.gnu.org/software/make/) available on your system if you aren't using a UNIX-based system for a better developer experience.
+   If you don't want to use `make` then please refer to the [Makefile](./Makefile) for specific commands on a given make target.
 
-Clone the source code from BentoML's GitHub repository:
-```bash
-git clone https://github.com/bentoml/BentoML.git && cd BentoML
-```
+2. Fork the BentoML project on [GitHub](https://github.com/bentoml/BentoML).
 
-Install BentoML with pip in `editable` mode:
-```bash
-pip install -e .
-```
+3. Clone the source code from your fork of BentoML's GitHub repository:
+   ```bash
+   git clone https://github.com/yourusername/BentoML.git && cd BentoML
+   ```
 
-This will make `bentoml` available on your system which links to the sources of
-your local clone and pick up changes you made locally.
+4. Add the BentoML upstream remote to your local BentoML clone:
+   ```bash
+   git remote add upstream git@github.com:bentoml/BentoML.git
+   ```
```

Comment:
I prefer not to encapsulate this under a dropdown details.

Reply:
that should be ok since it's for beginners only and they probably only read it once. But I can see people come back fro the rest of the docs during development